src/repl.ts

```diff
@@ -285,9 +299,12 @@ function _eval(service: Service, state: EvalState, input: string) {
     state.output = output;
   }
 
-  return changes.reduce((result, change) => {
-    return change.added ? exec(change.value, state.path) : result;
-  }, undefined);
+  return {
+    awaitPromise,
+    result: changes.reduce((result, change) => {
+      return change.added ? exec(change.value, state.path) : result;
```

Comment:
Can we wrap `change.value` in `processTopLevelAwait` and avoid calling it elsewhere?